openpype/hosts/maya/plugins/create/create_animation_pointcache.py

```diff
@@ -0,0 +1,289 @@
+        )
+    )
+
+    abc_defs.append(
+        TextDef(
+            "attr",
+            label="Custom Attributes",
+            default=getattr(cls, "attr", None),
+            placeholder="attr1; attr2; ...",
+            disabled=True if "attr" not in abc_args_overrides else False,
+        )
+    )
+
+    abc_defs.append(
+        TextDef(
+            "attrPrefix",
+            label="Custom Attributes Prefix",
+            default=getattr(cls, "attrPrefix", None),
+            placeholder="prefix1; prefix2; ...",
+            disabled=True if "attrPrefix" not in abc_args_overrides else False,
+        )
+    )
+
+    abc_defs.append(
+        EnumDef(
+            "dataFormat",
+            label="Data Format",
+            default=getattr(cls, "dataFormat", None),
+            items=["ogawa", "HDF"],
+            disabled=True if "dataFormat" not in abc_args_overrides else False,
+        )
+    )
+
+    abc_defs.append(
+        NumberDef(
+            "preRollStartFrame",
+            label="Start frame for preroll",
+            default=getattr(cls, "preRollStartFrame", None),
+            tooltip=(
+                "The frame to start scene evaluation at. This is used to set"
+                " the starting frame for time dependent translations and can"
+                " be used to evaluate run-up that isn't actually translated."
+            ),
+            disabled=True
+            if "preRollStartFrame" not in abc_args_overrides
+            else False,
+        )
+    )
+
+    return abc_defs
+
+
+def _ensure_defaults(cls, instance_data):
+    """Ensure we get default values when an attribute is not overrideable.
+
+    In instances where an attribute used to be modifiable, and then was locked
+    again, we want to make sure that we pass the default (what's on the
+    settings) instead of any value that might have been stored in the scene
+    when the attribute was modifiable.
+    """
+    abc_args_overrides = getattr(cls, "abc_args_overrides", [])
+    creator_attr = instance_data["creator_attributes"]
+    attr_default = getattr(cls, "attr", "")
+
+    if "attr" not in abc_args_overrides:
+        creator_attr["attr"] = attr_default
+
+    if "attrPrefix" not in abc_args_overrides:
+        creator_attr["attrPrefix"] = getattr(cls, "attrPrefix", "")
+
+    if "dataFormat" not in abc_args_overrides:
+        creator_attr["dataFormat"] = getattr(cls, "dataFormat", "")
+
+    if "preRollStartFrame" not in abc_args_overrides:
+        creator_attr["preRollStartFrame"] = getattr(
+            cls, "preRollStartFrame", ""
+        )
+
+    (
+        abc_boolean_defaults,
+        abc_boolean_overrides,
+    ) = _get_alembic_boolean_arguments(cls)
+
+    creator_attr["abcDefaultExportBooleanArguments"] = abc_boolean_defaults
+
+    for idx, arg in enumerate(creator_attr["abcExportBooleanArguments"]):
+        if arg not in abc_boolean_overrides:
+            del creator_attr["abcExportBooleanArguments"][idx]
```

Comment:
This is faulty logic - you are removing indices while you're iterating the list without making a copy of the list

Also the lookup here should really be against a `set` for best performance.

```python
abc_boolean_overrides = set(abc_boolean_overrides)
creator_attrs["abcExportBooleanArguments"] = [
    arg for arg in creator_attrs["abcExportBooleanArguments"]
    if arg not in abc_boolean_overrides
]